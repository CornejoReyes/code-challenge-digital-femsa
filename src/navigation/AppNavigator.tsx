import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { Dashboard } from '../features/dashboard';
import { Product } from '../features/products';
import type { RootStackParamList } from './types';

const { Navigator, Screen } = createNativeStackNavigator<RootStackParamList>();

export default function AppNavigator() {
  return (
    <Navigator screenOptions={{ headerShown: false }}>
      <Screen name="Dashboard" component={Dashboard} />
      <Screen name="ProductDetails" component={Product} />
    </Navigator>
  );
}
