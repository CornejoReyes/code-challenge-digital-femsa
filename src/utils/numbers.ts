export function formatNumberAsQuantity(value: number): string {
  const formatter = new Intl.NumberFormat('es-MX', { minimumFractionDigits: 2, maximumFractionDigits: 2 });
  return formatter.format(value);
}
