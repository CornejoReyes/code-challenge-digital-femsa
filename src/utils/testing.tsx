import { render, RenderOptions } from '@testing-library/react-native';
import { ReactElement } from 'react';
import Providers from '../Providers';

function customRender(ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) {
  return render(ui, { wrapper: Providers, ...options });
}

export * from '@testing-library/react-native';
export { customRender as render };
