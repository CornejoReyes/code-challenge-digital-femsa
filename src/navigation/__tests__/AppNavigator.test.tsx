import { render, screen } from '../../utils/testing';
import AppNavigator from '../AppNavigator';

describe('<AppNavigator />', () => {
  it('renders correctly', () => {
    render(<AppNavigator />);
    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getByText(/bienvenido/i)).toBeOnTheScreen();
  });
});
