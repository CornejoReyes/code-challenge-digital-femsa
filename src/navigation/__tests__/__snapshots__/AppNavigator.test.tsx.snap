// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavigator /> renders correctly 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(28, 28, 30)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Dashboard"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            dataSet={{}}
            style={
              {
                "flex": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Bienvenido de vuelta!
              </Text>
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Fernando Cornejo
              </Text>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "marginBottom": 24,
                  "marginTop": 24,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#71717a",
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 18,
                    "marginBottom": 16,
                    "textDecorationLine": undefined,
                  }
                }
              >
                TUS PUNTOS
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "backgroundColor": "#334FFA",
                    "borderRadius": 16,
                    "elevation": 5,
                    "marginLeft": "auto",
                    "marginRight": "auto",
                    "paddingBottom": 48,
                    "paddingTop": 48,
                    "position": "relative",
                    "shadowColor": "#000000",
                    "shadowOffset": {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 3.84,
                    "width": "80%",
                  }
                }
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#FFFFFF",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "left": 16,
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "position": "absolute",
                      "textDecorationLine": undefined,
                      "top": 16,
                    }
                  }
                >
                  Abril
                </Text>
                <View
                  accessibilityLabel="spinner container"
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <ActivityIndicator
                    accessibilityLabel="loading"
                    accessible={true}
                    color="#FFFFFF"
                    size="large"
                    style={
                      [
                        [
                          {},
                          {
                            "dataSet": {},
                          },
                          {},
                        ],
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#71717a",
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 18,
                    "marginBottom": 16,
                    "textDecorationLine": undefined,
                  }
                }
              >
                TUS MOVIMIENTOS
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 8,
                    "flex": 1,
                    "marginBottom": 16,
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                  }
                }
              >
                <RCTScrollView
                  data={[]}
                  getItem={[Function]}
                  getItemCount={[Function]}
                  keyExtractor={[Function]}
                  onContentSizeChange={[Function]}
                  onLayout={[Function]}
                  onMomentumScrollBegin={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  removeClippedSubviews={true}
                  renderItem={[Function]}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={[]}
                  viewabilityConfigCallbackPairs={[]}
                >
                  <View />
                </RCTScrollView>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityLabel="accumulated cta"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  colorScheme="primary"
                  dataSet={{}}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#334FFA",
                      "borderRadius": 8,
                      "elevation": 5,
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 10,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 10,
                      "shadowColor": "#000000",
                      "shadowOffset": {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                    test={true}
                  >
                    <View
                      dataSet={{}}
                      style={{}}
                    >
                      <Text
                        dataSet={{}}
                        style={
                          {
                            "backgroundColor": undefined,
                            "color": "#FFFFFF",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Ganados
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 16,
                    }
                  }
                />
                <View
                  accessibilityLabel="redeemed cta"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  colorScheme="primary"
                  dataSet={{}}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#334FFA",
                      "borderRadius": 8,
                      "elevation": 5,
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 10,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 10,
                      "shadowColor": "#000000",
                      "shadowOffset": {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                    test={true}
                  >
                    <View
                      dataSet={{}}
                      style={{}}
                    >
                      <Text
                        dataSet={{}}
                        style={
                          {
                            "backgroundColor": undefined,
                            "color": "#FFFFFF",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Canjeados
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</View>
`;
